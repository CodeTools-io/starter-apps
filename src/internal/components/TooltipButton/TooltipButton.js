import React, { useRef, useState } from 'react';
import { Box, Button, Drop } from 'grommet';

export default function TooltipButton({
  align = { bottom: 'top' },
  dropProps = {},
  icon,
  tooltip,
  tooltipProps = {
    pad: 'xsmall',
    background: 'TooltipButtonBackgroundColor',
  },
  ...props
}) {
  const [over, setOver] = useState(false);
  const ref = useRef();

  return (
    <Box>
      <Button
        ref={ref}
        onMouseOver={() => setOver(true)}
        onMouseLeave={() => setOver(false)}
        plain
        {...props}
      >
        <Box pad={{ vertical: 'small' }} align="center">
          {icon}
        </Box>
      </Button>
      {ref.current && over && (
        <Drop
          align={align}
          target={ref.current}
          plain
          trapFocus={false}
          {...dropProps}
        >
          <Box {...tooltipProps}>{tooltip}</Box>
        </Drop>
      )}
    </Box>
  );
}
