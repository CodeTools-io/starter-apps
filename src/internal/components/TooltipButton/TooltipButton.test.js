import React from 'react'
import TooltipButton from './TooltipButton'

describe('TooltipButton', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <TooltipButton>
        <p>Content for the TooltipButton</p>
      </TooltipButton>
    )

    expect(container).toMatchSnapshot()
  })
})

