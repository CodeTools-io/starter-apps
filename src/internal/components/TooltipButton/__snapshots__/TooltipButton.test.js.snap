// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipButton can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 gAsOLC"
  >
    <button
      class="StyledButton-sc-323bzc-0 kUDdtj"
      type="button"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fEOApw"
      />
    </button>
  </div>
</div>
`;
